File: 0623-Add One Row to Tree/0623-Add One Row to Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* addOneRow(TreeNode* root, int v, int d) {
        if (d == 1)
        {
            TreeNode* node = new TreeNode(v);
            node->left = root;
            return node;
        }
        
        helper(root, v, d);
        return root;
    }
    
private:
    void helper(TreeNode* root, int v, int d) {
        if (root == nullptr)
        {
            return;
        }
        if (d == 2)
        {
            TreeNode* nleft = new TreeNode(v);
            TreeNode* nright = new TreeNode(v);
            nleft->left = root->left;
            nright->right = root->right;
            root->left = nleft;
            root->right = nright;
        }
        else
        {
            helper(root->left, v, d - 1);
            helper(root->right, v, d - 1);
        }
    }
};

File: 0601-0700/0623-Add One Row to Tree/0623-Add One Row to Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def addOneRow(self, root: TreeNode, v: int, d: int) -> TreeNode:
        if d == 1:
            node = TreeNode(v)
            node.left = root
            return node
        if root is None:
            return root
        if d == 2:
            left = TreeNode(v)
            right = TreeNode(v)
            left.left = root.left
            right.right = root.right
            root.left = left
            root.right = right
        else:
            self.addOneRow(root.left, v, d - 1)
            self.addOneRow(root.right, v, d - 1)
        return root
